fenix/app/src/main/java/org/mozilla/fenix/settings/search/SaveSearchEngineFragment.kt

```diff
@@ -0,0 +1,200 @@
+import org.mozilla.fenix.ext.requireComponents
+import org.mozilla.fenix.ext.showToolbar
+import org.mozilla.fenix.settings.SupportUtils
+
+/**
+ * A [Fragment] that allows user to add a new search engine.
+ */
+@SuppressWarnings("LargeClass", "TooManyFunctions")
+class SaveSearchEngineFragment : Fragment(R.layout.fragment_save_search_engine) {
+
+    private val args by navArgs<SaveSearchEngineFragmentArgs>()
+    private var searchEngine: SearchEngine? = null
+
+    private var _binding: FragmentSaveSearchEngineBinding? = null
+    private val binding get() = _binding!!
+
+    private val inputListener = object : TextWatcher {
+        override fun afterTextChanged(editable: Editable) {
+            val bothFieldsHaveInput = binding.editEngineName.text?.isNotBlank() == true &&
+                binding.editSearchString.text?.isNotBlank() == true
+            binding.saveButton.isEnabled = bothFieldsHaveInput
+        }
+
+        override fun beforeTextChanged(
+            s: CharSequence?,
+            start: Int,
+            count: Int,
+            after: Int,
+        ) = Unit
+
+        override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) =
+            Unit
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+
+        searchEngine = requireComponents.core.store.state.search.customSearchEngines.find { engine ->
+            engine.id == args.searchEngineIdentifier
+        }
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+
+        _binding = FragmentSaveSearchEngineBinding.bind(view)
+
+        binding.saveButton.apply {
+            isEnabled = false
+            setOnClickListener { createCustomEngine() }
+        }
+
+        binding.editEngineName.addTextChangedListener(inputListener)
+        binding.editSearchString.addTextChangedListener(inputListener)
+
+        searchEngine?.let {
+            val url = it.resultUrls[0]
+            binding.editEngineName.setText(it.name)
+            binding.editSearchString.setText(url.toEditableUrl())
+        }
+
+        binding.customSearchEnginesLearnMoreWrapper.setOnClickListener {
+            (activity as HomeActivity).openToBrowserAndLoad(
+                searchTermOrURL = SupportUtils.getSumoURLForTopic(
+                    requireContext(),
+                    SupportUtils.SumoTopic.CUSTOM_SEARCH_ENGINES,
+                ),
+                newTab = true,
+                from = BrowserDirection.FromSaveSearchEngineFragment,
+            )
+        }
+    }
+
+    override fun onResume() {
+        super.onResume()
+        if (args.searchEngineIdentifier != null) {
+            showToolbar(getString(R.string.search_engine_edit_custom_search_engine_title))
+        } else {
+            showToolbar(getString(R.string.search_engine_add_custom_search_engine_title))
```

Comment:
Does this toolbar code need to run in `onResume` specifically? I think we could avoid introducing another function and do it in in `onViewCreated`

Reply:
I was just copying `AddSearchEngineFragment`, but not taking a closer look `showToolbar` is mostly called during `onResume` call throughout the app.

Reply:
Okay, no need to reinvent the wheel then